wgpu-core/src/command/render.rs

```
@@ -189,7 +189,7 @@ pub struct RenderPassDescriptor<'a> {
 pub struct RenderPass {
     base: BasePass<RenderCommand>,
     parent_id: id::CommandEncoderId,
-    color_targets: ArrayVec<Option<RenderPassColorAttachment>, { hal::MAX_COLOR_ATTACHMENTS }>,
+    color_targets: Vec<Option<RenderPassColorAttachment>>,
```

If we check the condition, we should be able to keep it an arrayvec

---

Currently RenderPass's all validation occurs within `command_encoder_run_render_pass_impl`, unless this validation is placed separately in `wgc::command::RenderPass::new()`

---

How about this: we keep it an ArrayVec, silently fail to push if we run over, then keep a side tally of how many it actually tried to enqueue (with a note as to why we had to do it this way)

---

I'm sorry, I don't quite understand how to do "keep a side tally", add a field to `BasePass` to record it? (only `BasePass` can be accessed in `trace::Command::RunRenderPass`)

---

Yeah we sould need to keep a count - why can't we put it in the count in the renderpass directly?